lib/ruby_lsp/requests/support/sorbet.rb

```diff
@@ -10,97 +10,68 @@ class << self
 
           ANNOTATIONS = T.let(
             {
-              "abstract!" => Annotation.new(arity: 0),
-              "absurd" => Annotation.new(arity: 1, receiver: true),
-              "all" => Annotation.new(arity: (2..), receiver: true),
-              "any" => Annotation.new(arity: (2..), receiver: true),
-              "assert_type!" => Annotation.new(arity: 2, receiver: true),
-              "attached_class" => Annotation.new(arity: 0, receiver: true),
-              "bind" => Annotation.new(arity: 2, receiver: true),
-              "cast" => Annotation.new(arity: 2, receiver: true),
-              "class_of" => Annotation.new(arity: 1, receiver: true),
-              "enums" => Annotation.new(arity: 0),
-              "interface!" => Annotation.new(arity: 0),
-              "let" => Annotation.new(arity: 2, receiver: true),
-              "mixes_in_class_methods" => Annotation.new(arity: 1),
-              "must" => Annotation.new(arity: 1, receiver: true),
-              "must_because" => Annotation.new(arity: 1, receiver: true),
-              "nilable" => Annotation.new(arity: 1, receiver: true),
-              "noreturn" => Annotation.new(arity: 0, receiver: true),
-              "requires_ancestor" => Annotation.new(arity: 0),
-              "reveal_type" => Annotation.new(arity: 1, receiver: true),
-              "sealed!" => Annotation.new(arity: 0),
-              "self_type" => Annotation.new(arity: 0, receiver: true),
-              "sig" => Annotation.new(arity: 0),
-              "type_member" => Annotation.new(arity: (0..1)),
-              "type_template" => Annotation.new(arity: 0),
-              "unsafe" => Annotation.new(arity: 1),
-              "untyped" => Annotation.new(arity: 0, receiver: true),
+              "abstract!" => { arity: 0 },
```

Comment:
I think I would still like to keep these are `Annotation` nodes, but redefine `Annotation` in this file as: `Annotation = Struct.new(:arity, :receiver, keyword_init: true)` (or as a `T::Struct`).

This would make it easier to type the `ANNOTATIONS` has and again make it possible to omit the `receiver` argument in the constructor (since `nil` is falsy). Moreover, it would change all the `[:arity]` accesses (which could return `nil`, btw, if there was a typo), into `.arity` accesses, which would be more type-safe.